src/App.tsx

```diff
@@ -1,14 +1,48 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useMemo, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { FilterBy } from './types/FilterBy';
+import { getFilteredTodos } from './utils/getFilteredTodos';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasRejectResponse, setHasRejectResponse] = useState(false);
+  const [query, setQuery] = useState('');
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.ALL);
+
+  useEffect(() => {
+    const onLoadGetTodos = async () => {
+      try {
+        const allTodos = await getTodos();
+
+        setTodos(allTodos);
+      } catch (error) {
+        setHasRejectResponse(true);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    onLoadGetTodos();
+  }, []);
+
+  const visibleTodos = useMemo(() => (
+    getFilteredTodos(todos, filterBy, query)
+  ), [query, filterBy, todos]);
+
+  const hasError = !isLoading && hasRejectResponse;
+  const hasNoError = !isLoading && !hasRejectResponse;
```

Comment:
Why don't you use one of these variables with `!` in conditional rendering, they are almost the same

Reply:
@alexandra-protyanova that would mean that we will change isLoading back to state `true`, and then during loading I will see also an empty List at the same time.

Reply:
agree, skip comment